### DISCUSSION TITLE
APM

### DISCUSSION THREAD

**Discussion by ElectronicsArchiver** [atom-ide-community/atom#383] [Q&A] [answered]
Did I [see that right](https://github.com/atom/apm/issues/951) or is APM / atom.io going down with Atom as well?

➞ Alternative?

**Comment by DeeDeeG** [comment, marked as answer]
It's true the API server is returning `500 internal server error`. Other than that, no-one knows what will happen to official atom.io.

The current workaround is to try hitting the atom.io/packages API a second time and hopefully not get a 500 internal server error the second/third time, etc. So if you are trying to `apm publish` and it doesn't go through, then manually undo the changes to your repo, and try again. Eventually it should work.

Long term, there is an effort to replace the functionality of the API server and website with our own. Feel free to contribute: 
- https://github.com/confused-Techie/atom-community-server-backend-JS
- https://github.com/confused-Techie/AtomPackagesArchive

The sunset announcement happened, and to my knowledge there has been no comments from the official side of Atom or past team members since then. I considered reaching out to past contributors, but I wonder if any of them know, since they're only past team members. I have hesitated to reach out because I don't want to put them in an awkward position, or broadcast their usernames and make them targets of harassment in the worst case scenario. My issue at upstream had no comments on it from maintainers, nor have the issues at the apm repo. I don't think we'll get official answers at this rate, but I haven't tried (while risking being annoying) directly pinging past contributors yet. That could be done if it comes down to it.

**Reply by ElectronicsArchiver** [reply]
Will your projects be adopted into Atom-Community or is it gonna be separate?

**Reply by AlexWayfer** [reply]
I already see archived packages by great maintainers: https://github.com/wooorm/linter-remark/pull/15#issuecomment-1159483043

**Reply by confused-Techie** [reply]
> Will your projects be adopted into Atom-Community or is it gonna be separate?

Its a good question if they will be merged with Atom-Community. The archiver will not, since its only use is to backup the data currently available on Atom.io. Otherwise the Server is looking to be the backend for Atom-Community, as I have been working alongside the other maintainers on this repo while building it. Even if it is not, the URL for the server will likely be set in the Atom Core for this repo.

**Reply by DeeDeeG** [reply]
@AlexWayfer for what it's worth, "archived" repositories/packages, that aren't deleted or set to private, should still provide the tarballs that apm needs in order to install those packages.

**Reply by AlexWayfer** [reply]
> @AlexWayfer for what it's worth, "archived" repositories/packages, that aren't deleted or set to private, should still provide the tarballs that apm needs in order to install those packages.

Yeah, but I mean maintaining, especially for big changes like ES6 modules migration.
